altair/utils/schemapi.py

```diff
@@ -181,19 +240,53 @@ def __str__(self):
             for val in schema_path[:-1]
             if val not in (0, "properties", "additionalProperties", "patternProperties")
         )
-        message = self.message
-        if self._additional_errors:
-            message += "\n        " + "\n        ".join(
-                [e.message for e in self._additional_errors]
-            )
-        return """Invalid specification
 
-        {}, validating {!r}
+        # Output all existing parameters when an unknown parameter is specified
+        if (
+            hasattr(vegalite, schema_path.split(".")[-1])
+            and self.validator == "additionalProperties"
+        ):
+            altair_class = schema_path.split(".")[-1]
```

Comment:
I shortened this to remove the `altair` prefix in the error message and just say e.g. <code>\`X\`</code> instead of `altair.X`

Reply:
I've no idea if this is possible or desirable, but as this becomes something like this:
```
See the help for `Scale` to read the full description of these parameters
```
Could this also become a hyperlink to the docs? See also this [snippet](https://github.com/altair-viz/altair/pull/2610/files#diff-d110e8643c6feff37c22c1a5c7d213b8384177718b8e26edd8e1b8f2ed027a0aR545-R548) that I saw in https://github.com/altair-viz/altair/pull/2610:
```python
# Patch in a reference to the class this docstring is copied from,
# to generate a hyperlink
doclines = Obj.__doc__.splitlines()
doclines[0] = f"Refer to :class:`{Obj.__name__}`"
```

Reply:
Hmm it is possible to put in the full link, but there is not way to hide it behind a friendlier text since the error message only outputs raw unformatted text (I believe). We would have to dump out the full URL and it does get pretty long, e.g.  `https://altair-viz.github.io/user_guide/generated/core/altair.Encoding.html#altair.Encoding`.

I actually considered removing this line completely and I am still not sure how helpful it is. I added it to encourage people to read the docstring of the class inside their IDE, but it is not always helpful as the name someone might have used in their spec does not always match with the one in the error message. For example, if I do `.mark_point(asdf='fdsa')` the error message will point the user to `MarkDef` rather than `mark_point` which might be unclear for people new to altair. I am not sure if it is worth improving that though.

cc @binste for thoughts here as well

Reply:
SQLAlchemy somewhat solved the issue with long URLs with a URL shortener which they host at `sqlalche.me`. See https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/exc.py#L60 how they then craft the URL in the exception message. Don't know what service is behind it and how we could keep something like that up-to-date. Maybe there is a good Sphinx extension which helps? I'm also in favour of not including the long URLs and maybe exploring the URL shortener or other ideas in a future release.

No strong preference if that line should be dropped. I see the point that references to classes such as `MarkDef` might be more confusing then they help and implementing some special logic is maybe also one for a future release.

Reply:
I vote for punting this to the future and opened https://github.com/altair-viz/altair/issues/2910 with a reference to this discussion.